src/tests/Common/CLRTest.Jit.targets

```diff
@@ -55,60 +55,22 @@ export RunningIlasmRoundTrip=
 ]]>
       </IlasmRoundTripBashScript>
 
-      <IlasmRoundTripBashScript Condition="'$(CLRTestKind)' == 'BuildAndRun'"><![CDATA[
+      <IlasmRoundTripBashScript Condition="'$(CLRTestKind)' == 'BuildAndRun' And '$(IlasmRoundTripIncompatible)' != 'true'"><![CDATA[
 # IlasmRoundTrip Script
-# Disable Ilasm round-tripping for Linker tests.
-# Todo: Ilasm round-trip on linked binaries.
 
 if [ -z "$DoLink" -a ! -z "$RunningIlasmRoundTrip" ];
 then
-  mkdir IL-RT
-  cp $(InputAssemblyName) $(TargetAssemblyName)
-  echo "$CORE_ROOT/ildasm" -raweh -unicode -out=$(DisassemblyName) $(TargetAssemblyName)
-  "$CORE_ROOT/ildasm" -raweh -unicode -out=$(DisassemblyName) $(TargetAssemblyName)
+  python3 $(AssemblyName)_ilasmroundtrip.py
   ERRORLEVEL=$?
   if [ $ERRORLEVEL -ne 0 ]
   then
-    echo EXECUTION OF ILDASM - FAILED $ERRORLEVEL
-    exit 1
-  fi
-  if [ ! -f "$(DisassemblyName)" ]
-  then
-    echo "EXECUTION OF ILDASM - FAILED $(DisassemblyName) is missing"
-    exit 1
-  fi
-
-  ilasm_count=1
-  while true
-  do
-    echo "$CORE_ROOT/ilasm" -output=$(InputAssemblyName) $(_IlasmSwitches) $(DisassemblyName)
-    "$CORE_ROOT/ilasm" -output=$(InputAssemblyName) $(_IlasmSwitches) $(DisassemblyName)
-    ERRORLEVEL=$?
-    if [ $ERRORLEVEL -eq 0 ]
-    then
-      break
-    fi
-
-    echo EXECUTION $ilasm_count OF ILASM - FAILED with $ERRORLEVEL
-
-    if [ $ilasm_count -eq 3 ]
-    then
-      break
-    fi
-
-    ((ilasm_count++))
-    echo Trying again
-    sleep 10
-  done
-
-  if [ $ERRORLEVEL -ne 0 ]
-  then
+    echo ILASM ROUND-TRIP - FAILED $ERRORLEVEL
     exit 1
   fi
 fi
 ]]>
       </IlasmRoundTripBashScript>
-      <BashIlrtTestLaunchCmds Condition="'$(CLRTestKind)' == 'BuildAndRun'"><![CDATA[
+      <BashIlrtTestLaunchCmds Condition="'$(CLRTestKind)' == 'BuildAndRun' And '$(IlasmRoundTripIncompatible)' != 'true'"><![CDATA[
```

Comment:
This comment is more about the original IL-RT change (and any change shouldn't be part of this PR).

The ilasm roundtrip scripts already had a somewhat peculiar structure where they execute the original binary -and- the roundtripped binary.  Presumably this was to detect the case where the input binary was already broken, though any normal test run would already catch that.  The downside was that roundtrip testing had the cost of executing the tests twice.  However, now that the original binary is overwritten, I think it is just running the roundtripped binary twice.  On top of that, I think it will run the merged test wrapper script twice (for all tests) and then individual RequiresProcessIsolation tests twice within each cmd/sh invocation, so they might be called four times.

Reply:
I would be surprised if it ran the tests twice. I think how the tests get run is that it finds all `.cmd`/`.sh` files and runs them.

Reply:
I checked the logs.  For example, in [this log](https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-93368-merge-13a9ee7cff6f474f9b/Regression_2/1/console.9dbafa52.log?helixlogtype=result), the line

```
"C:\h\w\BADE09EA\p\corerun.exe" -p "System.Reflection.Metadata.MetadataUpdater.IsSupported=false" -p "System.Runtime.Serialization.EnableUnsafeBinaryFormatterSerialization=true"  Runtime_91839.dll
```

appears 4 times in two pairs.  It doesn't matter how the .cmd are found.  The wrapper .cmd itself runs the whole set of tests twice, and the individual test .cmd runs its test twice.

Reply:
Why do they run multiple times? Is this intended behavior?

Reply:
My first comment has what I know.  It looks like it was originally intended, but it's not useful now that the two runs are the same.  I don't think it's worth trying to recreate the original behavior.  Instead I think you can just delete the execution step in the ilasm code and rely on the "normal" execution.

Reply:
Ah I understand now. You are saying we could just delete the extra execution step since we already overwrite it the original DLL.

Reply:
Yea, this would be a good change to do.